x-pack/plugins/maps/public/components/toolbar_overlay/view.js

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiPopover,
+  EuiContextMenu,
+  EuiSelectable,
+  EuiHighlight,
+  EuiTextColor,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { getIndexPatternsFromIds } from '../../index_pattern_util';
+import _ from 'lodash';
+import { DRAW_TYPE } from '../../actions/store_actions';
+import { ES_GEO_FIELD_TYPE } from '../../../common/constants';
+
+const RESET_STATE = {
+  isPopoverOpen: false,
+  drawType: null
+};
+
+export class ToolbarOverlay extends React.Component {
+
+
+  state = {
+    isPopoverOpen: false,
+    uniqueIndexPatternsAndGeoFields: [],
+    drawType: null
+  };
+
+  _openToolbar = () => {
+    if (!this._isMounted) {
+      return;
+    }
+    this.setState(prevState => ({
+      isPopoverOpen: !prevState.isPopoverOpen,
+      drawType: null
+    }));
+  };
+
+  _closePopover = () => {
+    if (!this._isMounted) {
+      return;
+    }
+    this.setState(RESET_STATE);
+  };
+
+  _onIndexPatternSelection = (options) => {
+    if (!this._isMounted) {
+      return;
+    }
+
+    const selection = options.find((option) => option.checked);
+    const drawType = this.state.drawType;
+    this.setState(RESET_STATE, () => {
+      if (drawType) {
+        this.props.initiateDraw({ drawType: drawType, ...selection.value });
+      }
+    });
+  };
+
+  _activateDrawForFirstIndexPattern = (drawType) => {
+    if (!this._isMounted) {
+      return;
+    }
+    const indexPatternAndGeofield = this.state.uniqueIndexPatternsAndGeoFields[0];
+    this.setState(RESET_STATE, () => {
+      this.props.initiateDraw({ drawType: drawType, ...indexPatternAndGeofield });
+    });
+  };
+
+  _showIndexPatternSelection =  (drawType) => {
```

Comment:
This method is not needed. It is only called in a synchronous manor so the check for isMounted is not needed. Then the only thing the method does is set drawType state. So you can remove method and just call set state where this method is used